src/components/Editor/EditorInput/EditorInput.js

```diff
@@ -0,0 +1,110 @@
+import React, { createRef, useEffect, useRef, useState } from 'react'
+import { DragDropContext } from '@hello-pangea/dnd'
+import { useDispatch, useSelector } from 'react-redux'
+import { closeFile, setFocussedFileIndex, setOpenFiles } from '../EditorSlice'
+import { TabPanel, Tabs } from 'react-tabs'
+import Button from '../../Button/Button'
+import { CloseIcon } from '../../../Icons'
+import EditorPanel from '../EditorPanel/EditorPanel'
+import DraggableTab from '../DraggableTabs/DraggableTab'
+import DroppableTabList from '../DraggableTabs/DroppableTabList'
+import RunBar from '../../RunButton/RunBar'
+
+import './EditorInput.scss'
+
+const EditorInput = () => {
+  const project = useSelector((state) => state.editor.project)
+  const openFiles = useSelector((state) => state.editor.openFiles)
+  const focussedFileIndices = useSelector((state) => state.editor.focussedFileIndices)
+  const dispatch = useDispatch()
+
+  const onDragStart = (input) => {
+    const { source } = input
+    dispatch(setFocussedFileIndex({panelIndex: parseInt(source.droppableId), fileIndex: source.index}))
+  }
+
+  const onDragEnd = (result) => {
+    if(!result.destination) return
```

Comment:
for some reason my brain is focussed on these things today, haha, but if you drop this a line down you're assigning destination anyway, this is also a little more defensive in an _admittedly odd_ situation where result isn't a thing

Reply:
so I think the approach is good, I'd just drop it to line 29, and say `if (!destination) return;` to make best use of the destructuring

Reply:
Okay, have gone with that 👍